pkg/apis/workspaces/v1alpha2/buildGuidances.go

```diff
@@ -0,0 +1,51 @@
+package v1alpha2
+
+// +k8s:openapi-gen=true
+// +union
+type BuildGuidance struct {
+
+	// +optional
+	// Allows specifying a dockerfile to initiate build
+	Dockerfile *Dockerfile `json:"dockerfile,omitempty"`
+
+	// +optional
+	// Allows specifying a builder image to initiate s2i (SourceToImage) build
+	SourceToImage *SourceToImage `json:"s2i,omitempty"`
+}
+
+type Dockerfile struct {
+
+	// Name that allows referencing a build guidance
+	Name string `json:"name"`
```

Comment:
In terms of overriding does it makes sense move the name to the upper level, like we do with components:
```
components:
  - name: maven
    container: ....
```
? I'm not sure about just a question

Reply:
The name is mandatory at the upper-level to identity elements in a list. When the field is the patchMergeKey of the list.
Here there is only 1 build guidance in a project afaik.

However I see you defined a union @wtam2018 Are we sure that the 2 values (dockerfile or s2i) are really mutually-exclusive ?

Reply:
@davidfestal I'll have union removed dockerfile and s2i are not mutually excluded.   I should probably rename guidance to its plural form.

Reply:
@davidfestal @sleshchenko Based on the feedbacks from our previous meeting, build guidance should be similar and at the same level as component/command.  I will rework this and take build guidance out of project.